Orange/widgets/widget.py

```diff
@@ -233,6 +226,21 @@ def __init__(self, *args, **kwargs):
         """__init__s are called in __new__; don't call them from here"""
 
     @classmethod
+    def get_widget_description(cls):
+        if not cls.name:
+            return
+        properties = {name: getattr(cls, name) for name in
+                      ("name", "icon", "description", "priority", "keywords",
+                       "help", "help_ref", "url",
+                       "version", "background", "replaces")}
+        return WidgetDescription(
```

Comment:
How do you feel about returning a simple dict here and instantiating it in `WidgetDescription.from_module()`: `cls(**w.get_widget_description())`? This entangles the implementation even less and lets canvas itself fill its internal structures.